locations/Snakefile

```diff
@@ -73,17 +86,81 @@ rule download_state_tracts_boundaries:
+    output:
+        "data/puma/tl_{vintage}_{state_fips_code}_puma10.zip"
+    shell:
+        """
+        wget -P data/puma/ \
+             -N https://www2.census.gov/geo/tiger/TIGER{wildcards.vintage:q}/PUMA/tl_{wildcards.vintage:q}_{wildcards.state_fips_code:q}_puma10.zip
+        """
+
+rule download_state_pumas_boundaries:
+    message: "Downloading {wildcards.vintage} Census PUMA cartographic boundaries for {wildcards.state_fips_code}"
+    output:
+        "data/puma/cb_{vintage}_{state_fips_code}_puma10_500k.zip"
+    shell:
+        """
+        wget -P data/puma/ \
+             -N https://www2.census.gov/geo/tiger/GENZ{wildcards.vintage:q}/shp/cb_{wildcards.vintage:q}_{wildcards.state_fips_code:q}_puma10_500k.zip
+        """
+
+rule download_community_reporting_areas:
+    message: "Downloading Community reporting areas(CRAs) for Seattle"
+    output:
+        "data/cra/seattle_cra.geojson"
+    shell:
+        """
+        curl -o data/cra/seattle_cra.geojson \
+             'https://gisdata.seattle.gov/server/rest/services/COS/CommunityReportingAreas/MapServer/0/query?where=1%3D1&outFields=*&outSR=4326&f=json'
+        """
+
 # Import locations
 rule import:
-    message: "Importing Census tracts into ID3C"
+    message: "Importing geographies into ID3C"
+    input:
+        expand("log/import/{vintage}_{state.fips_code}_tract", vintage = vintages, state = states_with_tracts.itertuples()),
+        expand("log/import/{vintage}_{state.fips_code}_puma", vintage = vintages, state = states_with_pumas.itertuples()),
+        "log/import/seattle_neighborhood_district"
+
+rule hierarchy:
+    message: "Creating tract hierarchy CSV file"
     input:
-        expand("log/import/{vintage}_{state.fips_code}_tract", vintage = vintages, state = states.itertuples())
+        puma = "data/2010_Census_Tract_to_2010_PUMA.txt",
+        neighborhood = "data/SeattleCensusBlocksandNeighborhoodCorrelationFile.xlsx"
+    output:
+        csv = temp("data/2010_Census_Tract_Hierarchy.csv")
+    run:
+        puma_hierarchies = (
+            pandas
+            .read_csv(input.puma, dtype = object)
+            .assign(
+                feature_identifier = lambda df: df.STATEFP + df.COUNTYFP + df.TRACTCE,
+                puma  = lambda df: df.STATEFP + df.PUMA5CE,
+                state = lambda df: df.STATEFP.apply(lambda x: hierarchy_state(fips_to_name(x))),
+                country = "united_states"
+            )
+        )
+        neighborhood_hierarchies = (
+            pandas
+            .read_excel(input.neighborhood, dtype = object)
+            .assign(
+                feature_identifier = lambda df: df.GEOID10.apply(lambda x: x[:-4]),
+                neighborhood_district = lambda df: df.NEIGHBORHOOD_DISTRICT_NAME.apply(lambda x: hierarchy_neighborhood(x))
+            )
+        )
+
+        hierarchies = puma_hierarchies.merge(neighborhood_hierarchies, on="feature_identifier", how="outer")
```

Comment:
I think something weird is happening in this join, but it might be elsewhere. The resulting CSV has lots of duplicate rows for Washington, e.g.:

![image](https://user-images.githubusercontent.com/79913/73695522-2730d200-468f-11ea-9163-b8575ad327ff.png)

Reply:
This is probably due to the `SeattleCensusBlocksandNeighborhoodCorrelationFile.xlsx` file. The file contains all census **blocks** within Seattle, so there are repeated rows for census tracts.